src/regional_settings.cpp

```diff
@@ -34,6 +34,51 @@ void read_and_set_or_throw( const JsonObject &jo, const std::string &member, T &
     }
 }
 
+namespace
+{
+generic_factory<om_settings_forest> om_forest_factory( "om_settings_forest" );
+} // namespace
+
+template<>
+const om_settings_forest &string_id<om_settings_forest>::obj() const
+{
+    return om_forest_factory.obj( *this );
+}
+template<>
+bool string_id<om_settings_forest>::is_valid() const
+{
+    return om_forest_factory.is_valid( *this );
+}
+void om_settings_forest::load_om_settings_forest( const JsonObject &jo, const std::string &src )
+{
+    om_forest_factory.load( jo, src );
+}
+void om_settings_forest::load( const JsonObject &jo, const std::string_view )
+{
+    // if these settings aren't loaded, we just revert to the defaults.
+    optional( jo, was_loaded, "noise_threshold_forest", noise_threshold_forest );
+    optional( jo, was_loaded, "noise_threshold_forest_thick", noise_threshold_forest_thick );
+    optional( jo, was_loaded, "noise_threshold_swamp_adjacent_water",
+              noise_threshold_swamp_adjacent_water );
+    optional( jo, was_loaded, "noise_threshold_swamp_isolated", noise_threshold_swamp_isolated );
+    optional( jo, was_loaded, "river_floodplain_buffer_distance_min",
+              river_floodplain_buffer_distance_min );
+    optional( jo, was_loaded, "river_floodplain_buffer_distance_max",
+              river_floodplain_buffer_distance_max );
+}
+const std::vector<om_settings_forest> &om_settings_forest::get_all()
+{
+    return om_forest_factory.get_all();
+}
+void om_settings_forest::reset_om_settings_forest()
+{
+    om_forest_factory.reset();
+}
+bool om_settings_forest::is_valid() const
+{
+    return om_forest_factory.is_valid( this->id );
```

Comment:
this isn't exactly what this function is for. i don't even think this is the right way to call validation on the string id, and i'm not sure the benefit of calling it through the template class since you're already overloading the is_valid method for om_settings_forest

Reply:
This should be `string_id<om_settings_forest>::is_valid` - you care if the object associated with the id exists, so if this is a function of the object it will always be true.